wlauto/workloads/androbench/__init__.py

```diff
@@ -0,0 +1,84 @@
+#    Copyright 2013-2015 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import os
+import re
+import time
+
+from wlauto import AndroidBenchmark
+from uiautomator import Device
+
+
+class Androbench(AndroidBenchmark):
+
+    name = 'androbench'
+    description = """Androbench measures the storage performance of device"""
+    package = 'com.andromeda.androbench2'
+    activity = '.main'
+    device=''
+
+    def setup(self, context):
+		global device
+		os.system('adb devices > deviceinfo')
+		devinf = open('deviceinfo','rb')
+		dev = devinf.readlines()[1].split('\t')[0]
+		devinf.close()
+		device=Device(dev)
+		os.system('rm deviceinfo')
+
+
+    def run(self, context):
+		global device,package,activity
+		os.system('adb shell pm clear com.andromeda.androbench2')		
+		os.system('adb shell am start -n com.andromeda.androbench2/.main')
+		while True :
+			if device(text="Measure your storage performance").exists :
+				time.sleep(1)
+				break
+				
+		if device(textStartsWith="Micro").exists :
+			device(textStartsWith="Micro").click()
+		if device(text="Yes").exists :
+			device(textStartsWith="Yes").click()
+		
+		
+		while True :
+			if device(text="Cancel").exists :
+				device(text="Cancel").click()
+				time.sleep(1)
+				break
+			
+
+    def update_result(self, context):
+        super(Androbench, self).update_result(context)
+        os.system('adb shell cp /data/data/com.andromeda.androbench2/databases/history.db /sdcard/results.db')
+        os.system('adb pull /sdcard/results.db .')
```

Comment:
Pelase use the WA Device interface (instantiated as `self.device`) to interract with the device. Do not invoke adb directly. Also locations on the host should generally be specifed relative to `context.output_directory`; do not create files in the current working directory. The above two lines should be relace with something like

    device_results = self.device.path.join(self.device.package_data_directory, self.package, 'databases', 'history.db')
    host_results = os.path.join(context.output_directory, 'results.db')
    self.device.pull_file(device_results, host_results, as_root=True)

(the `as_root=True` will take care of the intermediate transfer via the SD card.)